src/ai/backend/manager/actions/action/batch.py

```diff
@@ -0,0 +1,32 @@
+from abc import ABC, abstractmethod
+from typing import TypeVar
+
+from ai.backend.manager.data.permission.parameters import (
+    BatchQueryParams,
+)
+
+from .base import BaseAction
+
+
+class BaseBatchAction(BaseAction, ABC):
+    @abstractmethod
+    def target_entity_ids(self) -> list[str]:
+        """Return the IDs of the entities this action operates on."""
+        raise NotImplementedError
+
+    def permission_query_params(self) -> BatchQueryParams:
+        return BatchQueryParams(
+            entity_type=self.entity_type(),
+            operation_type=self.operation_type(),
+            entity_ids=self.target_entity_ids(),
+        )
```

Comment:
Let's make it return an empty value for entity_id so that it's not added to other actions before migration.